sphinx/application.py

```diff
@@ -134,7 +136,7 @@ class Sphinx:
     :ivar outdir: Directory for storing build documents.
     """
 
-    warningiserror: bool
+    warningiserror: Final = False
```

Comment:
```suggestion
    warningiserror: Final[bool] = False
```

Small nitpick.

Reply:
`Final` with a literal will take the type of that literal -- meaning we don't need to do `Final[Literal[False]]` here